src/IntegerConversion.hpp

```diff
@@ -0,0 +1,150 @@
+/*
+ *~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ * Copyright (c) 2019, Lawrence Livermore National Security, LLC.
+ *
+ * Produced at the Lawrence Livermore National Laboratory
+ *
+ * LLNL-CODE-746361
+ *
+ * All rights reserved. See COPYRIGHT for details.
+ *
+ * This file is part of the GEOSX Simulation Framework.
+ *
+ * GEOSX is a free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License (as published by the
+ * Free Software Foundation) version 2.1 dated February 1999.
+ *~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ */
+
+/**
+ * @file IntegerConversion.hpp
+ */
+
+#ifndef INTEGERCONVERSION_HPP_
+#define INTEGERCONVERSION_HPP_
+
+// Source includes
+#include "StringUtilities.hpp"
+#include "Macros.hpp"
+
+// System includes
+#include <limits>
+#include <typeinfo>
+
+/**
+ * @file IntegerConversion.hpp
+ */
+
+/**
+ * @brief function to take an unsigned integer and convert to a signed integer
+ * @tparam RTYPE the type of signed integer to convert into
+ * @tparam T     the type of the unsigned integer provided
+ * @param input  the integer to convert
+ * @return the converted integer
+ *
+ * This function takes in an unsigned integer and converts it to a signed integer. There is a check
+ * to make sure that no data is lost as a result of this conversion.
+ */
+template< typename RTYPE, typename T >
+inline typename std::enable_if< std::is_unsigned< T >::value && std::is_signed< RTYPE >::value,
+                                RTYPE >::type
+integer_conversion( T input )
```

Comment:
is this actually a new file?

Reply:
No. I think if you look commit by commit you'll see that it got moved but that when GitHub merges all the changes together it thinks it's new.

Reply:
No. If you look commit by commit it should show that it was moved but when GitHub displays all the changes at once it things its a new file.